app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -204,11 +210,11 @@ def pulp3_url(path = '/pulp/api/v3')
       end
 
       def pulp_mirror?
-        self.has_feature? PULP_NODE_FEATURE
+        self.has_feature?(PULP_NODE_FEATURE) || (self.has_feature?(SmartProxy::PULP3_FEATURE) && self.setting(SmartProxy::PULP3_FEATURE, 'mirror'))
       end
 
       def pulp_master?
-        self.has_feature? PULP_FEATURE
+        self.has_feature?(PULP_FEATURE) || (self.has_feature?(SmartProxy::PULP3_FEATURE) && !self.setting(SmartProxy::PULP3_FEATURE, 'mirror'))
```

Comment:
I think that when a setting exists, implies it has the feature. If a feature is removed, there are no settings stored. So I think the `has_feature?` is redundant if you check `mirror` is `false`.

Reply:
Yeah, but we're also trying to differentiate between not having the feature and having the feature set to false.   If it doesn't have the feature its not a pulp_master, if the mirror setting is false its not a pulp_master, so i think it makes sense to differentiate here.  

On the 'pulp_mirror?' function we could do that though and just check that the setting is true.  I'll update that!

Reply:
Don't we always set the value? There is a default (https://github.com/theforeman/smart_proxy_pulp/blob/3eca0c86a3414fb5e8fdeb29dfd918d2a39f0ffa/lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb#L8) so isn't this the same?
```suggestion
        self.has_feature?(PULP_FEATURE) || self.setting(SmartProxy::PULP3_FEATURE, 'mirror') == false
```

Reply:
oh you mean an explicit check on == false.  That works for me